webui: add "delete all conversations" button to import/export tab

thomasjfox opened the pull request:
- Add 'Delete all conversations' functionality with confirmation dialog
- Add Trash icon and destructive styling for clear visual indication
- Include success summary showing count of deleted conversations

@allozaur let me know what you think about the general approach here.

While testing I noticed I need to clear the current conversion. This is done using chatStore.clearActiveConversation(). A different approach would be to redirect to "?new_chat=true#/" to show the fancy banner.

This is my first time working with TypeScript or Svelte. As a C++ / Rust systems programmer, I appreciate your patience. :grin: Feel free to edit the code any way you like.

UPDATE: Demo video of the improved version:

![webui-delete-all-conversations](https://github.com/user-attachments/assets/092a27f3-ada5-4b82-a839-2d17365518cc)

allozaur commented on tools/server/webui/src/lib/components/app/chat/ChatSettings/ChatSettingsImportExportTab.svelte:
```diff
@@ -149,6 +155,50 @@
+			showDeleteDialog = true;
+		} catch (err) {
+			console.error('Failed to load conversations for deletion:', err);
+			alert('Failed to load conversations');
```
Please use existing Toast Notification component that we already are using :)

allozaur reviewed (changes requested):
Few minor tweaks needed :) also it'd be great if you attached a short video demo in the PR description.

Last, but not least, after we've got the PR approved, i will ask you to push the new static build, but let's first wait to make sure the PR is ready code-wise.

thomasjfox committed:
webui: add "delete all conversations" button to import/export tab

- Add 'Delete all conversations' functionality with confirmation dialog
- Add Trash icon and destructive styling for clear visual indication
- Redirects to "?new_chat=true#/" by using conversationsStore.deleteAll()

thomasjfox commented:
Two small things from my side:

1. Is it safe that conversationsStore.deleteConversation() triggers await goto(`?new_chat=true#/`)
  while further conversations will be deleted by handleDeleteAllConfirm()? In my testing all conversations were indeed deleted.  To phrase differently, is the goto() only executed by the browser when handleDeleteAllConfirm() is done?

2. I did rebuild the webui build, but I would not trust me, the random-stranger-from-the-Internet, to do that. ^^
Ideally an established team member does the builds and commits the minified JS-code. I've deployed the changed webui internally and it works, but who knows if my build environment is clean / set up properly. Just my two cents.

allozaur reviewed (changes requested):
> Two small things from my side:
> 
>     1. Is it safe that conversationsStore.deleteConversation() triggers await goto(`?new_chat=true#/`)
>        while further conversations will be deleted by handleDeleteAllConfirm()? In my testing all conversations were indeed deleted.  To phrase differently, is the goto() only executed by the browser when handleDeleteAllConfirm() is done?

We have this `goto` only if we are on an active conversation route:

```ts
			if (this.activeConversation?.id === convId) {
				this.activeConversation = null;
				this.activeMessages = [];
				await goto(`?new_chat=true#/`);
			}
```

But actually now taking a look at this i realize that instead of handling the delete all logic inside of the component, this should be sth like `conversationsStore.deleteAll()` method used from the UI component.

So please create this new method in the conversations store and then use it from the component level :)

>     2. I did rebuild the webui build, but I would not trust me, the random-stranger-from-the-Internet, to do that. ^^
>        Ideally an established team member does the builds and commits the minified JS-code. I've deployed the changed webui internally and it works, but who knows if my build environment is clean / set up properly. Just my two cents.

No worries, if you ran `npm run dev`, git hooks pass on pre-commit and `npm run build` works, it's all good ;)

thomasjfox committed:
chore: update webui build output
